cel/folding_test.go

```diff
@@ -279,15 +280,59 @@ func TestConstantFoldingOptimizer(t *testing.T) {
 			expr:   `1 + 2 + x ==  x + 2 + 1`,
 			folded: `3 + x == x + 2 + 1`,
 		},
+		{
+			expr:   `google.expr.proto3.test.ImportedGlobalEnum.IMPORT_BAR`,
+			folded: `1`,
+		},
+		{
+			expr:   `c == google.expr.proto3.test.ImportedGlobalEnum.IMPORT_BAZ ? "BAZ" : "Unknown"`,
+			folded: `"BAZ"`,
+		},
+		{
+			expr: `[
+						google.expr.proto3.test.ImportedGlobalEnum.IMPORT_BAR,
+						c,
+						google.expr.proto3.test.ImportedGlobalEnum.IMPORT_FOO
+					].exists(e, e == google.expr.proto3.test.ImportedGlobalEnum.IMPORT_FOO)
+						? "has Foo" : "no Foo"`,
+			folded: `"has Foo"`,
+		},
+		{
+			expr:   `l.exists(e, e == "foo") ? "has Foo" : "no Foo"`,
+			folded: `"has Foo"`,
+		},
+		{
+			expr:   `"foo" in l`,
+			folded: `true`,
+		},
+		{
+			expr:   `o.repeated_int32`,
+			folded: `[1, 2, 3]`,
+		},
 	}
 	e, err := NewEnv(
 		OptionalTypes(),
 		EnableMacroCallTracking(),
 		Types(&proto3pb.TestAllTypes{}),
-		Variable("x", DynType))
+		Variable("x", DynType),
+		Constant("c", IntType, types.Int(proto3pb.ImportedGlobalEnum_IMPORT_BAZ)),
```

Comment:
Would you mind adding a non-proto constant as well? Enums will be in the reference map, but it's worth seeing what this does with something that doesn't usually go into the reference map like a string value